test/functional/p2p_blocksonly.py

```diff
@@ -57,29 +56,35 @@ def run_test(self):
             self.nodes[0].p2p.wait_for_tx(txid)
             assert_equal(self.nodes[0].getmempoolinfo()['size'], 1)
 
-        self.log.info('Check that txs from forcerelay peers are not rejected and relayed to others')
-        self.log.info("Restarting node 0 with forcerelay permission and blocksonly")
-        self.restart_node(0, ["-persistmempool=0", "-whitelist=127.0.0.1", "-whitelistforcerelay", "-blocksonly"])
+        self.log.info('Check that txs from RELAY-whitelisted peers are not rejected and relayed to others')
+        self.log.info("Restarting node 0 with relay permission and blocksonly")
+        self.restart_node(0, ["-persistmempool=0", "-whitelist=127.0.0.1", "-whitelistrelay", "-blocksonly"])
```

Comment:
Can't this be written shorter as:

```suggestion
        self.restart_node(0, ["-persistmempool=0", "-whitelist=relay@127.0.0.1", "-blocksonly"])
```

Reply:
I thought so too, but interestingly it doesn't work.
When setting `whitelist=relay@127.0.0.1` and doing a `getpeerinfo`, the `permissions` are correctly set to `relay`, but `whitelisted` is set to `false`...

Reply:
`whitelisted` is there for backward compatibility only:
`// If this flag is present, the user probably expect that RPC and QT report it as whitelisted (backward compatibility)`.

Your test should still work even after marco's suggestion? You just have to update the following stuff too: drop the legacy `assert_equal(peer_1_info['whitelisted'], True)` stuff

Reply:
(I'm fine keeping it as is too)

Reply:
Great, thanks for the explanation, the rest of the test indeed works fine once changed.
I'll remove that check if that is a deprecated flag, it can be misleading for newcomers.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/19963/commits/f59681ecc422bbfe32d470ef96bbd25199ec14d0